src/Types.hs

```diff
@@ -1010,3 +1010,27 @@ countIdeaVotes v = countEq v ideaVoteValue
 
 countCommentVotes :: UpDown -> CommentVotes -> Int
 countCommentVotes v = countEq v commentVoteValue
+
+
+-- * event log
+
+data EventLog = EventLog IdeaSpace [(Timestamp, EventLogItem)]
+  deriving (Eq, Ord, Show, Read)
+
+data EventLogItem =
+    EventLogUserCreates           User Thing
+  | EventLogUserEdits             User Thing
+  | EventLogUserMarksIdeaFeasible User IdeaJuryResultValue Idea
+  | EventLogUserVotesOnIdea       User Idea IdeaVote
+  | EventLogUserVotesOnComment    User Idea CommentVote
+  | EventLogUserDelegates         User Delegation User
+  | EventLogTopicMovesFromTo      Phase Phase EventTriggeredBy
+  | EventLogIdeaMovesToTopic      User Idea Topic
+  | EventLogIdeaWins              User Idea
+  deriving (Eq, Ord, Show, Read)
+
+data EventTriggeredBy = EventTriggeredBy User | EventTriggeredByTimeout | EventTriggeredByAllIdeasMarked
+  deriving (Eq, Ord, Show, Read)
+
+data Thing = TopicThing Topic | IdeaThing Idea | CommentThing Comment
```

Comment:
I don't get the name `Thing` :) Could you rename it to a more descriptive name, or add a comment?